frontend/src/components/my-grant-profile-page-handler/my-grant.tsx

```diff
@@ -0,0 +1,225 @@
+import {
+  Button,
+  Grid,
+  GridColumn,
+  Link,
+  Spacer,
+  Text,
+  VerticalStack,
+} from "@python-italia/pycon-styleguide";
+import { FormattedMessage } from "react-intl";
+
+import { useCurrentLanguage } from "~/locale/context";
+import { DeadlineStatus, Status as GrantStatus } from "~/types";
+import type { MyProfileWithGrantQuery } from "~/types";
+
+import { useCountries } from "~/helpers/use-countries";
+import { createHref } from "../link";
+import { Sidebar } from "./sidebar";
+
+type Props = {
+  grant: MyProfileWithGrantQuery["me"]["grant"];
+  deadline: MyProfileWithGrantQuery["conference"]["deadline"];
+};
+
+export const MyGrant = ({ grant, deadline }: Props) => {
+  const language = useCurrentLanguage();
+  const countries = useCountries();
+
+  const canManageGrant = [
+    GrantStatus.WaitingForConfirmation,
+    GrantStatus.Confirmed,
+    GrantStatus.WaitingList,
+    GrantStatus.WaitingListMaybe,
+  ].includes(grant.status);
+
+  const getCountryLabel = (value: string): string | undefined => {
+    const country = countries.find((country) => country.value === value);
+    return country ? country.label : undefined;
+  };
+
+  const dateFormatter = new Intl.DateTimeFormat(language, {
+    day: "numeric",
+    month: "long",
+    year: "numeric",
+  });
+
+  return (
```

Comment:
can we split this into some sub-components? it is quite a big component

Reply:
I moved the table in a separate component. Do you think it's enough? or should I split also the "next steps" section maybe? 🤔

Reply:
this component is still a bit hard to read but can't think of anything specific now without cloning the code locally and changing it a bit